### ISSUE TITLE
Create one function or one pair of functions to replace all `_load_*_data` and `_process_*_data_df` functions

### ISSUE THREAD

**Issue by leewujung** [uw-echospace/EchoPro#130]
There are very minimal differences between the handling of all of those data types, which can be distinguished with an input argument `data_type`

I also don't see why it is absolutely necessary to set specific columns as `index` for the dataframes. The downstream processing I have seen so far are really just using those as regular variables, and the code would look cleaner without those being `index`, especially for those there are more than one in haul to transect mapping.

- Under `LoadBioData`
  - `_load_specimen_data` is identical to `_process_length_data_df` except for:
    - `df.sort_index(inplace=True)` -- which can be added to `_process_length_data_df` without problem (I suspect not having it in `_process_length_data_df` is actually a bug)
    - the section below:
      ```python
      # perform check on data
      if len(df["age"]) - df["age"].isna().sum() < 0.1 * len(df["age"]):
          raise RuntimeWarning("Aged data are less than 10%!\n")
      ```
  - `_process_specimen_data` is completely identical to `_process_length_data_df`
  - `_process_haul_to_transect_mapping_data` is different from `_process_catch_data` that it does not do `df = self._extract_target_species(df)` and does not apply haul offset
    - but in fact, haul offset is applied outside in `_load_haul_to_transect_mapping_data`
- Under `LoadStrateData`:
    - `_load_stratification_file` and `_load_geographic_stratification` are pretty much identical to the various `_process_*_data_df` plus `_load_*_data`

**Comment by emiliom** [comment]
Yeah. The simplifications I made recently to the `data_loader` modules have made those similarities / parallels more apparent, and the further simplifications you're suggesting (or most of them) easier to implement.

**Comment by leewujung** [comment]
This is addressed via #161.
